### PR TITLE
Merge dev-requirements into requirements

### PR DESCRIPTION
**Pull request by ruestefa**
Merge `dev-requirements.yml` into `requirements.yml` (and `dev-environment.yml` into `environment.yml`) because conda-env cannot properly handle requirements distributed over multiple YAML files (only the version restrictions in the last YAML file are guaranteed to be respected).

Moving the remaining two files to the project root has been moved to a separate PR (#93) in order to simplify `copier -f update`s of blueprint-based projects (see explanation there).

See #86 for some background on the requirements merge.

### CODE REVIEW

**Comment by ruestefa** [comment]
See https://github.com/MeteoSwiss-APN/mch-python-blueprint/issues/86#issuecomment-1376047824

**Comment by ruestefa** [comment]
launch jenkins

**Comment by ruestefa** [comment]
launch jenkins

**Comment by ruestefa** [comment]
Already contains #92

**Comment by regDaniel** [comment]
> Already contains #92

Perfect, I approved #92 and will proceed with the review of this one.

**Review by regDaniel — CHANGES_REQUESTED** [review]
We should rephrase the few sentences on the installation process in `tmpl/docs/installation.rst.j2`. Otherwise it looks fine to me.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -10,18 +10,16 @@ repos:
...
-                tmpl/|
-                docs/readme.rst|
-            )$
 -   repo: https://github.com/codespell-project/codespell
-    rev: v2.2.1
+    rev: v2.2.2
     hooks:
     -   id: codespell
+        additional_dependencies:
+        -   tomli
```
[the remark this answers is not in the document]
Comment (ruestefa) [reply .pre-commit-config.yaml:25]: Indeed, I've been annoyed by this, as well. I probably would've gone for the shorter `.yml`, but if the official recommendation is `.yaml`, then `.yaml` it shall be.

File: tmpl/docs/installation.rst.j2 [hunk 2]
```diff
@@ -13,11 +13,11 @@ To install {{ project_name }} you need a miniconda installation. You can either
...
+Dependencies are handled by the conda package manager. The goal of this step is to set up a conda environment according to the requirements of {{ project_name }}. Note that the development installation has some additional dependencies as it includes linters and other development tools. The dependencies are handled in requirement files. Free installations are based on the `requirements/requirements.yml` file, where the first-level dependencies of the package are listed. Pinned installations are based on exported environments and stored in the file `requirements/environment.yml`.yml`. In total, four possible installation options are possible pinned/unpinned x dev/prod. These options are covered in the script `setup_env.sh`. The optional flag `-u` stands for unpinned installation:
```
[the remark this answers is not in the document]
Comment (ruestefa) [reply tmpl/docs/installation.rst.j2:16]: Yeah, I didn't revise the instructions in full, I just searched for the file names and fixed the respective sentences...

**Comment by regDaniel** [comment]
launch jenkins

**Comment by ruestefa** [comment]
launch jenkins

**Review by regDaniel — APPROVED** [review]

**Comment by regDaniel** [comment]
launch jenkins

**Comment by regDaniel** [comment]
launch jenkins

**Pull request merged**